### PR TITLE
run_command: Add new kwarg 'capture'

### PR DESCRIPTION
**Pull request by nirbheek**
`capture: false` means we won't try to read the output at all.

Closes https://github.com/mesonbuild/meson/issues/3364

Will add docs and a release snippet if this is approved.

### CODE REVIEW

File: mesonbuild/interpreter.py [hunk 1]
```diff
@@ -104,12 +106,19 @@ def run_command(self, cmd, args, source_dir, build_dir, subdir, mesonintrospect,
             cwd = os.path.join(source_dir, subdir)
         child_env = os.environ.copy()
         child_env.update(env)
+        stdout = subprocess.PIPE
```
Comment (sarum9in) [inline mesonbuild/interpreter.py]: `stdout = subprocess.PIPE if self.capture else subprocess.DEVNULL`? Splitting initialization in several statements has a risk that one of them will be moved later.

**Comment by sarum9in** [comment]
Since you are already doing this, wouldn't it be better to use `capture_stdout`, `capture_stderr` instead? I am pretty sure there is a good chance we encounter exactly same problem for `stderr` later. And then we will have to introduce inconsistent keyword.

Similar question, why don't we implement `capture_stderr` now?

**Comment by nirbheek** [comment]
> Since you are already doing this, wouldn't it be better to use capture_stdout, capture_stderr instead? I am pretty sure there is a good chance we encounter exactly same problem for stderr later.

I don't know of any tool that outputs binary/garbage on `stderr`. If there are locale-specific text decoding issues for stderr, we should fix those. We don't want people to just disable things and move on. 

In summation, prematurely implementing features without a good user-story is a bad idea.

If we do find a good use for this, we can always implement `capture_stderr:` later, and possibly deprecate `capture:` in favour of a new `capture_stdout:`.

Also `capture:` matches the similar kwarg used in `custom_target()`.

File: mesonbuild/interpreter.py (hunk 1 above)
Comment (nirbheek) [reply mesonbuild/interpreter.py]: Sure.

File: mesonbuild/interpreter.py [hunk 2]
```diff
@@ -1740,6 +1747,7 @@ def get_cross_property_method(self, args, kwargs):
                     'project': {'version', 'meson_version', 'default_options', 'license', 'subproject_dir'},
                     'run_command': {'check'},
                     'run_target': {'command', 'depends'},
+                    'run_command': {'capture'},
```
Comment (ignatenkobrain) [inline mesonbuild/interpreter.py]: __[Flake8]__

[F601] dictionary key 'run_command' repeated with different values

**Comment by jpakkane** [comment]
An alternative solution for this would be to store `stdout` and `stderr` as raw binary clumps and only try to decode them if the getter functions are called and then error out. That's a fair bit of work, though, and I'm not sure if that is really needed by anyone.

If that solution is not deemed ok, then it's fine to merge this as is.

**Comment by eli-schwartz** [comment]
> Will add docs and a release snippet if this is approved.

We ended up with a release snippet but no docs. :/

**Pull request merged**